src/lightning/pytorch/loops/progress.py

```diff
@@ -124,7 +124,7 @@ class _Progress(_BaseProgress):
     current: _ReadyCompletedTracker = field(default_factory=_ProcessedTracker)
 
     def __post_init__(self) -> None:
-        if type(self.total) is not type(self.current):  # noqa: E721
+        if self.total.__class__ is not self.current.__class__:
```

Comment:
on master, we're failing with this strange error all of a sudden:
https://results.pre-commit.ci/run/github/178626720/1691069135.3AiReCsKQbiReN-0UeAdVg
can't add a noqa marker here because pre-commit will remove it